From fd4669c85c611dbb0513049ffec971400d6dfa49 Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Fri, 5 Jan 2024 14:22:35 +0100
Subject: [PATCH 0328/1695] dts: stm32: add default HCONF1 value for VDDIO on
 stm32mp257f-dk board

Add the default initialization values for stm32mp25 DK board.
Update the shadow default value to allow the system
to freeze the watchdog in stop or standby

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 438a915531c0ca643eff5a774c00f1c6b5637927 dts: stm32: add default HCONF1 value for VDDIO on stm32mp257f-dk board]
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Change-Id: Iabf45116fd08103a2e7fd4de817ebc10c8a28be2
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/349482
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp257f-dk.dts | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp257f-dk.dts b/core/arch/arm/dts/stm32mp257f-dk.dts
index 88cbf5a99..bbf55079b 100644
--- a/core/arch/arm/dts/stm32mp257f-dk.dts
+++ b/core/arch/arm/dts/stm32mp257f-dk.dts
@@ -6,6 +6,7 @@
 
 /dts-v1/;
 
+#include <dt-bindings/soc/stm32mp-provisioning.h>
 #include "stm32mp257.dtsi"
 #include "stm32mp25xf.dtsi"
 #include "stm32mp25-pinctrl.dtsi"
@@ -27,6 +28,14 @@
 		device_type = "memory";
 		reg = <0x0 0x80000000 0x1 0x00000000>;
 	};
+
+	shadow-prov {
+		compatible = "st,provisioning";
+		hconf1_prov {
+			nvmem-cells = <&hconf1_otp>;
+			st,shadow-value = <0x00018DB6>;
+		};
+	};
 };
 
 &risaf2 {
-- 
2.47.2

